peewee_async.py

```diff
@@ -1359,21 +1247,66 @@ class AioQueryMixin:
     async def aio_execute(self, database):
         return await database.aio_execute(self)
 
+    async def as_async_query_wrapper(self, cursor):
+        result = AsyncQueryWrapper(cursor=cursor, query=self)
+        await result.fetchall()
+        return result
+
 
 class AioModelDelete(peewee.ModelDelete, AioQueryMixin):
-    pass
+    async def fetch_results(self, cursor):
+        if self._returning:
+            return await self.as_async_query_wrapper(cursor=cursor)
+        return cursor.rowcount
 
 
 class AioModelUpdate(peewee.ModelUpdate, AioQueryMixin):
-    pass
+
+    async def fetch_results(self, cursor):
+        if self._returning:
+            return await self.as_async_query_wrapper(cursor=cursor)
+        return cursor.rowcount
 
 
 class AioModelInsert(peewee.ModelInsert, AioQueryMixin):
-    pass
+    async def fetch_results(self, cursor):
+        if self._returning is not None and len(self._returning) > 1:
+            return await self.as_async_query_wrapper(cursor=cursor)
+
+        if self._returning:
+            row = await cursor.fetchone()
+            if row is not None:
+                result = row[0]
+            else:
+                result = None
+        else:
+            last_id = await self._database.last_insert_id_async(cursor)
+            result = last_id
+
+        return result
 
 
 class AioModelSelect(peewee.ModelSelect, AioQueryMixin):
 
+    async def fetch_results(self, cursor):
+        return await self.as_async_query_wrapper(cursor)
+
+    @peewee.database_required
+    async def aio_scalar(self, database, as_tuple=False, as_dict=False):
```

Comment:
as_dict not used?

Reply:
I thought the deprecated scalar had had it either. Actually i copy-pasted it from sync peewee and i think we will add this param lately. But i will remove it for the current state. Thank you